gammapy/time/lightcurve.py

```diff
@@ -800,20 +804,31 @@ def compute_flux_point(self, time_interval, spectral_model, energy_range):
 
             flux = measured_excess / predicted_excess.value
             flux *= int_flux
-            flux_err = int_flux / predicted_excess.value
+
             # Gaussian errors, TODO: should be improved
-            flux_err *= excess_error(n_on=n_on, n_off=n_off, alpha=alpha_mean)
+            flux_err = int_flux / predicted_excess.value
+            delta_excess = excess_error(n_on=n_on, n_off=n_off, alpha=alpha_mean)
+            flux_err *= delta_excess
+
+            flux_ul = -1
+            sigma = significance_on_off(n_on=n_on, n_off=n_off, alpha=alpha_mean, method='lima')
+            if sigma <= sigma_ul_thres:
+                flux_ul = Helene_ULs(n_on - alpha_mean * n_off, delta_excess, 99.73) #3 sigma ULs
+                flux_ul *= int_flux / predicted_excess.value
         else:
             flux = 0
             flux_err = 0
+            flux_ul = -1
+            sigma = 0.
 
         # Store measurements in a dict and return that
         return useinterval, OrderedDict([
             ('time_min', Time(tmin, format='mjd')),
             ('time_max', Time(tmax, format='mjd')),
             ('flux', flux * u.Unit('1 / (s cm2)')),
             ('flux_err', flux_err * u.Unit('1 / (s cm2)')),
-
+            ('flux_ul', flux_ul * u.Unit('1 / (s cm2)')),
+            ('is_ul', sigma <= sigma_ul_thres)
```

Comment:
@bkhelifi - You're missing a comma here:
```
$ python setup.py test -V -t gammapy/time/tests/test_lightcurve.py
``
```
________________________________________________________________________ test_lightcurve_estimator ________________________________________________________________________

    @requires_data('gammapy-extra')
    @requires_dependency('scipy')
    def test_lightcurve_estimator():
        spec_extract = spec_extraction()
        lc_estimator = LightCurveEstimator(spec_extract)
    
        # param
        intervals = []
        for obs in spec_extract.obs_list:
            intervals.append([obs.events.time[0], obs.events.time[-1]])
    
        model = PowerLaw(
            index=2.3 * u.Unit(''),
            amplitude=3.4e-11 * u.Unit('1 / (cm2 s TeV)'),
            reference=1 * u.TeV,
        )
    
        lc = lc_estimator.light_curve(
            time_intervals=intervals,
            spectral_model=model,
>           energy_range=[0.5, 100] * u.TeV,
        )

gammapy/time/tests/test_lightcurve.py:206: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
gammapy/time/lightcurve.py:672: in light_curve
    useinterval, row = self.compute_flux_point(time_interval, spectral_model, energy_range, sigma_ul_thres)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <gammapy.time.lightcurve.LightCurveEstimator object at 0xb147e2da0>
time_interval = [<Time object: scale='tt' format='mjd' value=53343.92100685648>, <Time object: scale='tt' format='mjd' value=53343.94050190008>]
spectral_model = PowerLaw(), energy_range = <Quantity [  0.5, 100. ] TeV>, sigma_ul_thres = 3.0

    def compute_flux_point(self, time_interval, spectral_model, energy_range, sigma_ul_thres=3.):
        """Compute one flux point for one time interval.
    
            Parameters
            ----------
            time_interval : `~astropy.time.Time`
                Time interval (2-element array, or a tuple of Time objects)
            spectral_model : `~gammapy.spectrum.models.SpectralModel`
                Spectral model
            energy_range : `~astropy.units.Quantity`
                True energy range to evaluate integrated flux (true energy)
            sigma_ul_thres : float
                Threshold on detection significance to compute Flux Upper Limits
    
            Returns
            -------
            useinterval : bool
                Is True if the time_interval produce a valid flux point
            measurements : dict
                Dictionary with flux point measurement in the time interval
            """
        tmin, tmax = time_interval[0], time_interval[1]
        livetime = 0
        alpha_mean = 0.
        alpha_mean_backup = 0.
        measured_excess = 0
        predicted_excess = 0
        n_on = 0
        n_off = 0
        useinterval = False
    
        # Loop on observations
        for t_index, obs in enumerate(self.obs_list):
    
            # discard observations not matching the time interval
            obs_start = obs.events.observation_time_start
            obs_stop = obs.events.observation_time_end
            if (tmin < obs_start and tmax < obs_start) or (tmin > obs_stop):
                continue
            useinterval = True
            # get ON and OFF evt list
            on, off, spec, e_reco = self._create_and_filter_onofflists(t_index=t_index, energy_range=energy_range,
                                                                       interval=[tmin, tmax], extra=True)
            n_on_obs = len(on.table)
            n_off_obs = len(off.table)
    
            # compute effective livetime (for the interval)
            if tmin >= obs_start and tmax <= obs_stop:
                # interval included in obs
                livetime_to_add = (tmax - tmin).to('s')
            elif tmin >= obs_start and tmax >= obs_stop:
                # interval min above tstart from obs
                livetime_to_add = (obs_stop - tmin).to('s')
            elif tmin <= obs_start and tmax <= obs_stop:
                # interval min below tstart from obs
                livetime_to_add = (tmax - obs_start).to('s')
            elif tmin <= obs_start and tmax >= obs_stop:
                # obs included in interval
                livetime_to_add = (obs_stop - obs_start).to('s')
            else:
                livetime_to_add = 0 * u.s
    
            # Take into account dead time
            livetime_to_add *= (1. - obs.observation_dead_time_fraction)
    
            # Compute excess
            obs_measured_excess = n_on_obs - spec.alpha * n_off_obs
    
            # Compute the expected excess in the range given by the user
            # but must respect the energy threshold of the observation
            # (to match the energy range of the measured excess)
            # We use the effective livetime and the right energy threshold
            e_idx = np.where(np.logical_and.reduce(
                (e_reco >= spec.lo_threshold,  # threshold
                 e_reco <= spec.hi_threshold,  # threshold
                 e_reco >= energy_range[0],  # user
                 e_reco <= energy_range[-1])  # user
            ))[0]
            counts_predictor = CountsPredictor(
                livetime=livetime_to_add,
                aeff=spec.aeff,
                edisp=spec.edisp,
                model=spectral_model
            )
            counts_predictor.run()
            counts_predicted_excess = counts_predictor.npred.data.data[e_idx[:-1]]
    
            obs_predicted_excess = np.sum(counts_predicted_excess)
    
            # compute effective normalisation between ON/OFF (for the interval)
            livetime += livetime_to_add
            alpha_mean += spec.alpha * n_off_obs
            alpha_mean_backup += spec.alpha * livetime_to_add
            measured_excess += obs_measured_excess
            predicted_excess += obs_predicted_excess
            n_on += n_on_obs
            n_off += n_off_obs
    
        # Fill time interval information
        if useinterval:
            int_flux = spectral_model.integral(energy_range[0], energy_range[1])
    
            if n_off > 0.:
                alpha_mean /= n_off
            if livetime > 0.:
                alpha_mean_backup /= livetime
            if alpha_mean == 0.:  # use backup if necessary
                alpha_mean = alpha_mean_backup
    
            flux = measured_excess / predicted_excess.value
            flux *= int_flux
    
            # Gaussian errors, TODO: should be improved
            flux_err = int_flux / predicted_excess.value
            delta_excess = excess_error(n_on=n_on, n_off=n_off, alpha=alpha_mean)
            flux_err *= delta_excess
    
            flux_ul = -1
            sigma = significance_on_off(n_on=n_on, n_off=n_off, alpha=alpha_mean, method='lima')
            if sigma <= sigma_ul_thres:
                flux_ul = Helene_ULs(n_on - alpha_mean * n_off, delta_excess, 99.73) #3 sigma ULs
                flux_ul *= int_flux / predicted_excess.value
        else:
            flux = 0
            flux_err = 0
            flux_ul = -1
            sigma = 0.
    
        # Store measurements in a dict and return that
        return useinterval, OrderedDict([
            ('time_min', Time(tmin, format='mjd')),
            ('time_max', Time(tmax, format='mjd')),
            ('flux', flux * u.Unit('1 / (s cm2)')),
            ('flux_err', flux_err * u.Unit('1 / (s cm2)')),
            ('flux_ul', flux_ul * u.Unit('1 / (s cm2)')),
            ('is_ul', sigma <= sigma_ul_thres)
>           ('livetime', livetime * u.s),
            ('alpha', alpha_mean),
            ('n_on', n_on),
            ('n_off', n_off),
            ('measured_excess', measured_excess),
            ('expected_excess', predicted_excess),
        ])
E       TypeError: 'tuple' object is not callable

gammapy/time/lightcurve.py:832: TypeError
=================================================================== 1 failed, 16 passed in 6.83 seconds ===================================================================
```